pkg/controller/cni/cilium_reconciler.go

```diff
@@ -0,0 +1,215 @@
+package cni
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	anywherev1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
+	eksacluster "github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/controller"
+	"github.com/aws/eks-anywhere/pkg/controller/serverside"
+	"github.com/aws/eks-anywhere/pkg/executables"
+	"github.com/aws/eks-anywhere/pkg/networking/cilium"
+)
+
+const defaultRequeueTime = time.Second * 10
+
+type CiliumReconciler struct{}
+
+func (cr *CiliumReconciler) Reconcile(ctx context.Context, log logr.Logger, cluster *anywherev1.Cluster, client client.Client, specWithBundles *eksacluster.Spec) (controller.Result, error) {
+	log.Info("Applying CNI")
+	ciliumDS, err := getCiliumDS(ctx, client)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Deploying Cilium DS")
+			helm := executables.NewHelm(executables.NewExecutable("helm"), executables.WithInsecure())
+
+			ci := cilium.NewCilium(nil, helm)
+
+			ciliumSpec, err := ci.GenerateManifest(ctx, specWithBundles, []string{constants.CapvSystemNamespace})
+			if err != nil {
+				return controller.Result{}, err
+			}
+			if err := serverside.ReconcileYaml(ctx, client, ciliumSpec); err != nil {
+				return controller.Result{}, err
+			}
+			return controller.Result{}, err
+		}
+
+		return controller.Result{}, err
+	}
+
+	// upgrade cilium
+	log.Info("Upgrading Cilium")
+	needsUpgrade, err := ciliumNeedsUpgrade(ctx, log, client, specWithBundles)
+	if err != nil {
+		return controller.Result{}, err
+	}
+
+	if !needsUpgrade {
+		log.Info("Skipping Cilium")
+		return controller.Result{}, nil
+	}
+
+	helm := executables.NewHelm(executables.NewExecutable("helm"))
```

Comment:
can we build helm exec on a higher level and use it in this method instead of building twice?